### PR TITLE
docs: fix link to architectural overview

### PR DESCRIPTION
**Pull request by mickeypash**
The link in one of the examples was 404ing

### CODE REVIEW

File: examples/aws/terraform/README.md [hunk 1]
```diff
@@ -11,7 +11,7 @@ If you are planning on using our Terraform example in production, please referen
 
 We recommend familiarizing yourself with the following resources prior to reviewing our Terraform examples:
 
-- [Teleport Architecture](https://gravitational.com/teleport/docs/architecture/teleport_architecture_overview/)
+- [Teleport Architecture](https://gravitational.com/teleport/docs/architecture/)
```
Comment (benarent) [inline examples/aws/terraform/README.md:14]:
```suggestion
- [Teleport Architecture](https://goteleport.com/teleport/docs/architecture/overview/)
```

**Review by benarent — DISMISSED** [review]
Please accept suggestion. this points to our new domain.

**Review by webvictim — APPROVED** [review]

**Comment by webvictim** [comment]
@mickeypash Thanks for the PR!

**Review by klizhentas — APPROVED** [review]

**Comment by mickeypash** [comment]
Hello!
The change requested was auto-merged!
So I think we are good to go?

**Comment by klizhentas** [comment]
@mickeypash thanks for the reminder, I will merge it

**Pull request merged**